src/blocks/SignUpForm.js

```diff
@@ -36,6 +38,7 @@ const SignUpForm = ({ toggleForm }) => {
     }
     //signUp function imported from Functions.js
     signUp(firstName, lastName, password, email, role);
+    navigate('/');
```

Comment:
`signUp` is doing an asynchronous HTTP request, so you don't want to navigate until we make sure the HTTP request is completed. 

We can make `signUp` return a promise: 
```
return axios.post...
```
That way, we can do `signUp(firstName, lastName, password, email, role).then((response) => ...`

Alternatively, you can move the axios call to `apiService.js` and remove the code in `Functions.js` altogether, which probably is the more consistent and better approach.

Reply:
Jacob moved the  axios call to `apiSerivce.js` before I took this issue off his hands.